packages/astro/src/runtime/server/render/server-islands.ts

```diff
@@ -71,7 +71,7 @@ let data = {
 	slots: ${safeJsonStringify(renderedSlots)},
 };
 
-let response = await fetch('/_server-islands/${componentId}', {
+let response = await fetch('/_server-islands/${componentId}${result.trailingSlash === 'always' ? '/' : ''}', {
```

Comment:
How would `build.format` affect the result of this request? We don't use `build.format`  to determine routing do we? The purpose of that config is for determining where to write files during the build, but these are API endpoints that don't get built to files.

Reply:
It doesn't, but that function is more future proof because it checks all the variants of trailingSlash, which means that in the future we have more variants, we are automatically covered. It's better to use one single function everywhere instead of handling the single cases.

Reply:
But that function is specifically about the build. It's in `core/build/util.ts`. `build.format` doesn't have any effect on the URLs of server endpoints. It's only for routes that are generated in SSG. The immediate problem here is that `build.format` is not available inside of the runtime, I had to add `trailingSlash` to make that available. So I'd have to add `build.format` too, but it seems very incorrect to do so. 

Another option is to create a new function just about trailingSlash in a util file somewhere, but that also feels slightly silly giving how small it would be.

Agree about `base`, I'll add that.

Reply:
OK! I think we can merge the PR once we add `base` and test for handling it. Thank you!